src/wrapper/lts.ts

```
@@ -210,3 +240,201 @@ export interface AddressBook {
+    );
+  }
+
+  // A map where the key is the bucket ID and the value is a tuple of the resource address and
+  // amount.
+  let bucketAmounts: Record<string, [string, Decimal]> = {};
+
+  let feesLocked:
+    | {
+        account: string;
+        amount: Decimal;
+      }
+    | undefined = undefined;
+  let withdraws: Record<string, Record<string, Decimal>> = {};
+  let deposits: Record<string, Record<string, Decimal>> = {};
+
+  // Iterate over the instructions and resolve them
+  for (const instruction of instructions) {
+    switch (instruction.instruction) {
+      case Instruction.Kind.TakeFromWorktopByAmount:
+        const takeFromWorktopInstruction =
+          instruction as Instruction.TakeFromWorktopByAmount;
+
+        // Assuming that the bucket id is a string since this is what the LTS library produces and
+        // because the non-string IDs are currently bugged in Scrypto.
+        const bucketId = (
+          takeFromWorktopInstruction.intoBucket
+            .identifier as ManifestAstValue.String
+        ).value;
+        const resourceAddress =
+          takeFromWorktopInstruction.resourceAddress.address;
+        const amount = takeFromWorktopInstruction.amount.value;
+
+        bucketAmounts[bucketId] = [resourceAddress, amount];
+        break;
+
+      case Instruction.Kind.CallMethod:
+        const callMethodInstruction = instruction as Instruction.CallMethod;
+
+        // Cases we support:
+        // 1. Withdraw by amount
+        // 2. Deposit by amount
+        // 3. Lock fee
+
+        // TODO: Support withdraw_and_lock_fee when the simple builder supports it
+
+        // Case: Lock Fee
+        if (
+          callMethodInstruction.methodName.value === "lock_fee" &&
+          callMethodInstruction.arguments?.length === 1 &&
+          callMethodInstruction.arguments[0].type ===
+            ManifestAstValue.Kind.Decimal
+        ) {
+          let lockFeeAccount = callMethodInstruction.componentAddress.address;
```

We should probably throw an error if this is the second lock fee instruction because that would be non-conforming, and otherwise we'd return incorrect information back.

---

Yup you're right, will throw an exception if a second one is found